esx_service/cli/vmdkops_admin.py

```diff
@@ -930,11 +933,17 @@ def tenant_vm_ls_headers():
 def generate_tenant_vm_ls_rows(vms):
     """ Generate output for tenant vm ls command """
     rows = []
-    for vm in vms:
-        # vm has the format like this (vm_uuid)
-        uuid = vm
-        name = vmdk_utils.get_vm_name_by_uuid(uuid)
-        rows.append([uuid, name])
+    for vm_uuid, vm_name in vms:
+        name = vmdk_utils.get_vm_name_by_uuid(vm_uuid)
+        # If the VM name cannot be resolved then use one from db
+        # If it is not available from db then mark it as NOT_AVAILABLE
+        if not name:
+            if vm_name:
+                name = vm_name
+            else:
+                name = NOT_AVAILABLE
```

Comment:
if not name and vm_name:
name = vm_name
else
name = NOT

won't be correct, since NOT AVAILABLE is used only when vm name wasn't resolved by get_vm_name_by_uuid as well as it wasn't found in DB.